4070 object keys (#4094)
Codecademy/docs: content/javascript/concepts/objects/terms/keys/keys.md

---
Title: '.keys()'
Description: 'Extracts keys from an object and returns them as an array'
Subjects:
  - 'Web Development'
  - 'Computer Science'  
Tags:
  - 'Objects'
  - 'Methods'  
CatalogContent: 
  - 'introduction-to-javascript'
  - 'paths/front-end-engineer-career-path'
---

> The `.keys()` static method extracts keys from an object and returns them as an array. `.keys()` only returns the keys for the object's property and only for enumerable properties.

Reviewer: ```suggestion
The `.keys()` static method extracts keys from an object and returns them as an array. It only returns the keys for the object's property and only for enumerable properties.
```

> >**Note:** The order of the keys in the resulting array is not guaranteed to be the same as the order in which they were defined in the object.

Reviewer: ```suggestion
> **Note:** The order of the keys in the resulting array is not guaranteed to be the same as the order in which they were defined in the object.
```

## Syntax

The basic syntax is:

```pseudo
Object.keys(obj)
```

> 

Reviewer: Write a description for the `obj` parameter below the syntax block.

## Example

> Here's a simple example that extracts keys from an object and returns them as an array:

Reviewer: ```suggestion
Here's a simple example that extracts keys from an object and returns them as an array:

```
Remember to give a line space before and after a code block.

---

> Title: 'Object.keys()'

Reviewer 2: ```suggestion
Title: '.keys()'
```

Description: 'Extracts keys from an object and returns them as an array'
Subjects:
  - 'Web Development'
  - 'Computer Science'  
Tags:
  - 'Objects'
  - 'Methods'  
CatalogContent: 
  - 'introduction-to-javascript'
  - 'paths/front-end-engineer-career-path'
---

> The `Object.keys()` static method is used to extract keys from an object and returns them as an array. `Object.keys()` only returns the keys for the object's property
> and only for enumerable properties.

Reviewer 2: ```suggestion
The `.keys()` static method extracts keys from an object and returns them as an array. `.keys()` only returns the keys for the object's property and only for enumerable properties.
```

>**Note:** The order of the keys in the resulting array is not guaranteed to be the same as the order in which they were defined in the object.

## Syntax

The basic syntax is:

```pseudo
Object.keys(obj)
```

> 
> - `obj` - An object.

Reviewer 2: ```suggestion
```

## Example

> Here's a simple example:

Reviewer 2: Instead of this give a basic desc of what the reader can expect from the coming code snippet

```js
const myObject = {
  name: 'John',
  age: 25,
  city: 'London'
};
const keysArray = Object.keys(myObject);
console.log(keysArray);
```

> ### Output:

Reviewer 2: ```suggestion

The above example will give the following output:
```

> ```pseudo

Reviewer 2: instead of pseudo use the shell tag for output blocks

---
Title: '.keys()'
Description: 'Extracts keys from an object and returns them as an array'
Subjects:
  - 'Web Development'
  - 'Computer Science'  
Tags:
  - 'Objects'
  - 'Methods'  
CatalogContent: 
  - 'introduction-to-javascript'
  - 'paths/front-end-engineer-career-path'
---

The `.keys()` static method extracts keys from an object and returns them as an array. It only returns the keys for the object's property and only for enumerable properties.

> **Note:** The order of the keys in the resulting array is not guaranteed to be the same as the order in which they were defined in the object.

## Syntax

The basic syntax is:

```pseudo
Object.keys(obj)
```

> `obj`: An object

Reviewer: ```suggestion
- `obj`: An object from which the keys are extracted.
```
